File: app/src/main/java/antoninovitale/dropcodechallenge/details/BeerDetailsPresenter.java
package antoninovitale.dropcodechallenge.details;

import antoninovitale.dropcodechallenge.api.model.Beer;

/**
 * Created by antoninovitale on 02/09/2017.
 */
interface BeerDetailsPresenter {
    void setView(BeerDetailsView view);

    void onChanged(Beer beer);

    void onStatusClick(int position);

    void onMethodStatusClick(int position);

    void onMethodEnd(int position);

    void onMethodTimeElapsed(int position, long millis);

}

File: app/src/main/java/com/ninovitale/punkapi/app/details/BeerDetailsPresenter.kt
package com.ninovitale.punkapi.app.details

import com.ninovitale.punkapi.app.api.model.Beer

/**
 * Created by antoninovitale on 02/09/2017.
 */
interface BeerDetailsPresenter {
    fun setView(view: BeerDetailsView?)
    fun onChanged(beer: Beer)
    fun onStatusClick(position: Int)
    fun onMethodStatusClick(position: Int)
    fun onMethodEnd(position: Int)
    fun onMethodTimeElapsed(position: Int, millis: Long)
}